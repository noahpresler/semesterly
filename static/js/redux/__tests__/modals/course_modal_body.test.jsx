/**
Copyright (C) 2017 Semester.ly Technologies, LLC

Semester.ly is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Semester.ly is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
**/

import React from 'react';
import renderer from 'react-test-renderer';
import CourseModalBody from '../../ui/modals/course_modal_body';
import courseModalBodyFixture from '../../__fixtures__/course_modal_body.fixture';

it('CourseModalBody correctly renders', () => {
  const tree = renderer.create(
    <CourseModalBody {...courseModalBodyFixture} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
