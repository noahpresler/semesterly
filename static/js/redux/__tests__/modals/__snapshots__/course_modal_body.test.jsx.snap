// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseModalBody correctly renders 1`] = `
<div
  className="modal-body"
>
  <div
    className="cf"
  >
    <div
      className="col-3-16"
    >
      <div
        className="credits"
      >
        <h3>
          1
        </h3>
        <h4>
           credit
        </h4>
      </div>
      <div
        className="rating-module"
      >
        <h4>
          Average Course Rating
        </h4>
        <div
          className="sub-rating-wrapper"
        >
          <div
            className="star-ratings-sprite"
          >
            <span
              className="rating"
              style={
                Object {
                  "width": "NaN%",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="capacity"
      >
        <div
          className="capacity__tracker-text"
        >
          <span>
            1
            % of Seats Added on Semesterly
          </span>
        </div>
      </div>
      <div
        className="modal-module prerequisites"
      >
        <h3
          className="modal-module-header"
        >
          Prerequisites
        </h3>
        <p>
          
          <a
            className="course-link"
            href={null}
          >
            <span>
              ECO100Y1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#FD7473",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#b15150",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  ECO100Y1
                </span>
                <span>
                  Introduction to Economics
                </span>
              </span>
            </span>
          </a>
          (67%)/(
          <a
            className="course-link"
            href={null}
          >
            <span>
              ECO101H1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#36DEBB",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#259b82",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  ECO101H1
                </span>
                <span>
                  Principles of Microeconomics
                </span>
              </span>
            </span>
          </a>
          (63%), 
          <a
            className="course-link"
            href={null}
          >
            <span>
              ECO102H1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#C585DE",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#895d9b",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  ECO102H1
                </span>
                <span>
                  Principles of Macroeconomics
                </span>
              </span>
            </span>
          </a>
          (63%))/
          <a
            className="course-link"
            href={null}
          >
            <span>
              ECO105Y1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#D4DBC8",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#97A086",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  ECO105Y1
                </span>
                <span>
                  Principles of Economics for Non-Specialists
                </span>
              </span>
            </span>
          </a>
          (80%); 
          <a
            className="course-link"
            href={null}
          >
            <span>
              MAT133Y1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#A3F5F2",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#53ABA8",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  MAT133Y1
                </span>
                <span>
                  Calculus and Linear Algebra for Commerce
                </span>
              </span>
            </span>
          </a>
          /(
          <a
            className="course-link"
            href={null}
          >
            <span>
              MAT135H1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#E7F76D",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#6C7A89",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  MAT135H1
                </span>
                <span>
                  Calculus 1(A)
                </span>
              </span>
            </span>
          </a>
          , 
          <a
            className="course-link"
            href={null}
          >
            <span>
              MAT136H1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#4c7fd8",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#375994",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  MAT136H1
                </span>
                <span>
                  Calculus 1(B)
                </span>
              </span>
            </span>
          </a>
          )/
          <a
            className="course-link"
            href={null}
          >
            <span>
              MAT137Y1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#5CCCF2",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#408ea9",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  MAT137Y1
                </span>
                <span>
                  Calculus
                </span>
              </span>
            </span>
          </a>
          /
          <a
            className="course-link"
            href={null}
          >
            <span>
              MAT157Y1
            </span>
            <span
              className="course-link-tip"
              style={
                Object {
                  "backgroundColor": "#FFD462",
                }
              }
            >
              <span
                className="slot-bar"
                style={
                  Object {
                    "backgroundColor": "#b29444",
                  }
                }
              />
              <span
                className="course-link-content"
              >
                <span>
                  MAT157Y1
                </span>
                <span>
                  Analysis I
                </span>
              </span>
            </span>
          </a>
          
        </p>
      </div>
      <div
        className="modal-module areas"
      >
        <h3
          className="modal-module-header"
        >
          Program of Study Tags
        </h3>
        <p>
          None
        </p>
      </div>
      <div
        className="modal-module friends"
      >
        <h3
          className="modal-module-header"
        >
          Friends In This Course
        </h3>
        <div
          id="friends-wrapper"
        >
          <div
            className="friends__inner"
          >
            <div
              className="friend"
            >
              <div
                className="ms-friend"
                style={
                  Object {
                    "backgroundImage": "url(https://scontent.xx.fbcdn.net/v/t1.0-1/p50x50/10957589_10152725992049147_2815259779743523470_n.jpg?oh=85550d1771fec7b1524ac1d9303cd697&oe=598CCC9B)",
                  }
                }
              />
              <p
                title="Linda Margolius Presler"
              >
                Linda Margolius Presler
              </p>
            </div>
            <div
              className="friend"
            >
              <div
                className="ms-friend"
                style={
                  Object {
                    "backgroundImage": "url(https://scontent.xx.fbcdn.net/v/t1.0-1/c163.0.204.204/s50x50/12321414_113682579025974_6205266913524508520_n.jpg?oh=dfa9c89403c0179bd6b00e4afd300569&oe=599A0E58)",
                  }
                }
              />
              <p
                title="Mylittlewilliamyao Isalwaysontheprowl"
              >
                Mylittlewilliamyao Isalwaysontheprowl
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal-module friends"
      >
        <h3
          className="modal-module-header"
        >
          Friends Who Have Taken This Course
        </h3>
        <div
          id="friends-wrapper"
        >
          <div
            className="friends__inner"
          >
            <div
              className="friend"
            >
              <div
                className="ms-friend"
                style={
                  Object {
                    "backgroundImage": "url(https://scontent.xx.fbcdn.net/v/t1.0-1/p50x50/10957589_10152725992049147_2815259779743523470_n.jpg?oh=85550d1771fec7b1524ac1d9303cd697&oe=598CCC9B)",
                  }
                }
              />
              <p
                title="Linda Margolius Presler"
              >
                Linda Margolius Presler
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-8-16"
    >
      <h3
        className="modal-module-header"
      >
        Reactions
      </h3>
      <p>
        Check out your classmate's reactions – click an emoji to add your own opinion!
      </p>
      <div
        className="reactions-wrapper"
      >
        <div
          className="reactions"
        >
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="🔥"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f525.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Must Take
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="😍"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f60d.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Love It
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="💩"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f4a9.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Crap Class
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="😐"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f610.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                It's Aight
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="😴"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f634.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Boring
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="😩"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f629.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Hard
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="💯"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f4af.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Easy
              </span>
            </div>
          </div>
          <div
            className="swing reaction"
            onClick={[Function]}
          >
            <div
              className="emoji"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            >
              <img
                alt="🤓"
                className="emoji"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/72x72/1f913.png"
              />
            </div>
            <div
              className="action-container"
            >
              <div
                className="emoji-action"
              >
                <i
                  className="fa fa-plus"
                />
              </div>
              <div
                className="emoji-count"
              >
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              className="reaction-dropdown"
            >
              <div
                className="tip-border"
              />
              <div
                className="tip"
              />
              <span>
                Interesting
              </span>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          className="modal-module-header"
        >
          Course Description
        </h3>
        <p>
          The use of microeconomics to analyze a variety of issues from marketing and finance to organizational structure. Consumer preferences and behavior; demand, cost analysis and estimation; allocation of inputs, pricing and firm behavior under perfect and imperfect competition; game theory and public policy, including competition policy. Business cases are used to connect theory and practice and to highlight differences and similarities between economics and accounting, marketing and finance. This course is restricted to students in the Commerce programs.
        </p>
        
      </div>
      <div
        className="modal-module"
      >
        <h3
          className="modal-module-header"
        >
          Course Evaluations
        </h3>
        <div
          className="modal-entry course-evaluations "
        >
          <p
            className="empty-intro"
          >
            No course evaluations for this course yet.
          </p>
        </div>
      </div>
    </div>
    <div
      className="col-5-16 cf"
      id="modal-section-lists"
    >
      <div>
        <h3
          className="modal-module-header"
        >
           
          Lecture Sections
           
          <small>
            (Hover to see the section on your timetable)
          </small>
           
        </h3>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L0101
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            S Hussain
          </h5>
          <h6>
            <span
              className="green"
            >
              140 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              140
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L0201
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            S Hussain
          </h5>
          <h6>
            <span
              className="green"
            >
              130 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              130
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L0301
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            S Hussain
          </h5>
          <h6>
            <span
              className="green"
            >
              140 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              140
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L0401
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            S Hussain
          </h5>
          <h6>
            <span
              className="green"
            >
              130 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              130
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L0501
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            P Foltin
          </h5>
          <h6>
            <span
              className="green"
            >
              130 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              130
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
        <div
          className="modal-section"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <h4>
            <span>
              L5101
            </span>
            <i
              className="fa fa-calendar-check-o"
            />
          </h4>
          <h5>
            P Blanchenay
          </h5>
          <h6>
            <span
              className="green"
            >
              140 open
            </span>
            <span>
               / 
            </span>
            <span
              className="total-seats"
            >
              140
               seats
            </span>
          </h6>
          <i
            className="fa fa-lock"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
